io/zenoh-links/zenoh-link-quic/src/unicast.rs

```diff
@@ -276,8 +270,17 @@ impl LinkManagerUnicastTrait for LinkManagerUnicastQuic {
             .map_err(|e| zerror!("Cannot create a new QUIC listener on {addr}: {e}"))?;
         server_crypto.server_config.alpn_protocols =
             ALPN_QUIC_HTTP.iter().map(|&x| x.into()).collect();
-        let mut server_config =
-            quinn::ServerConfig::with_crypto(Arc::new(server_crypto.server_config));
+
+        // Install rustls provider
+        rustls::crypto::ring::default_provider()
+            .install_default()
+            .ok();
```

Comment:
I think it's better to return an error if this operation fails.
```suggestion
            .map_err(|err| zerror!("Error installing TLS crypto provider: {err}."))?;
```

Reply:
``` rust
...
static PROCESS_DEFAULT_PROVIDER: OnceBox<Arc<CryptoProvider>> = OnceBox::new();
...

pub fn install_default(self) -> Result<(), Box<Arc<Self>>> {
    PROCESS_DEFAULT_PROVIDER.set(Box::new(Arc::new(self)))
}
```

Reply:
After testing, maybe the warning shouldn't be added, because it will be printed whenever try to connect

Reply:
I think it's still safer to map and return the error. The behaviour you observed may actually be true for the current version of `quinn` and `rustls` but no guarantee is provided it will behave the same in future versions.
By simply converting and returning the error, everything will be handled by upper logic.

Reply:
Hi @Mallets
I agree that it is only true for the current version，but can't throw error here, because when listening on the quic and connecting to the quic endpoint at the same time, install_default() will inevitably be executed multiple times, and all but the first execution will fail. If exec install_default() in only one place, and it has to install before ClientConfig::builder() or ServerConfig::builder(). 
That can only be done before the program is initialized and the listener begins. This will lose the effect of lazy loading crypto providers

Reply:
Right, I didn't have the knowledge about the fact that `install_default` will succeed the first time and it will fail any subsequent attempt. In this case it's fine to ignore the error but I'd add the explanation in a comment in the code on why we are ignoring the error, i.e. the explanation you just provided in your comment :)

Beside this point, the PR LGTM.